lib/facter/resolvers/aix/partitions.rb

```diff
@@ -56,14 +56,14 @@ def populate_from_lslv(name)
           end
 
           def compute_size(info_hash)
-            physical_partitions = info_hash['PPs'].to_i
+            logical_partitions = info_hash['LPs'].to_i
             size_physical_partition = info_hash['PP SIZE']
             exp = if size_physical_partition[/mega/]
                     Facter::Util::Aix::InfoExtractor::MEGABYTES_EXPONENT
                   else
                     Facter::Util::Aix::InfoExtractor::GIGABYTES_EXPONENT
                   end
-            size_physical_partition.to_i * physical_partitions * exp
+            size_physical_partition.to_i * logical_partitions * exp
```

Comment:
I'm not sure I understand this part, don't we want to report on the physical partition size?

Reply:
If we want to report the size of the partition (file system) correctly, we need to only consider the number of logical partitions. For example a 2GB `/usr` file system resides on the logical volume `/dev/hd2` which has:
- a `PP Size` of 512MB
- 4 `LP`s (4x 512MB = 2048MB)
- 2 `copies` (1 mirror), therefor 2x 4 `LP`s = 8 `PP`s
- 8 `PP`s (2x 4x 512MB = 4096MB)
The partition/file system size should be reported as 2GB (4 `LP`s) and not 4GB (8 `PP`s). As long as logical volume has no additional copies (>1), we could obviously use either `PP`s or `LP`s. But for mirrored volumes we must use the number of `LP`s. Hope this makes  sense and explains it. For a different phrasing see also the [official IBM AIX Documentation](https://www.ibm.com/docs/en/aix/7.2?topic=lvsc-logical-partitions)

Reply:
Ah thanks for the example! So if we're mirroring, then from a logical perspective we have 2GB, but from physical perspective we have 4GB, and you're saying facter should report the former. That makes sense to me.